numpy/_core/numeric.py

```diff
@@ -2385,6 +2385,19 @@ def _array_equal_dispatcher(a1, a2, equal_nan=None):
     return (a1, a2)
 
 
+_no_nan_types = {
+    np.dtypes.BoolDType,
+    np.dtypes.Int8DType,
+    np.dtypes.Int16DType,
+    np.dtypes.Int32DType,
+    np.dtypes.Int64DType,
+}
```

Comment:
Strange that I cannot repro locally, but the try to import modules lazily seems to make things barf.  Importing `dtypes` lazily is useless (The C side has to already import it to fill it!), but I guess explicitly importing here will work...

Reply:
It is actually a bit annoying, because `np.longlong` or `np.cint` could fall through the cracks (right now), but will ignore that.